src/service.js

```diff
@@ -182,12 +183,23 @@ class Service {
       data[this.id] = id;
     }
 
-    return this.Model
-      .update(params.query, { $set: data }, options)
-      .lean(this.lean)
-      .exec()
-      .then(() => this._getOrFind(id, params))
-      .catch(errorHandler);
+    let promise;
+
+    // We need this shitty hack because update doesn't return
+    // a promise properly when runValidators is true. WTF!
+    try {
+      promise = this.Model
```

Comment:
Yeah I'll update so it just returns. Keeps it cleaner. It's not the same because we our `update` method because patch can support multiple objects whereas we said update is only good for 1 object. I'll double the tests to make sure we have one that ensures objects are correct after a patch.